packages/lib/src/interactions/SelectionTool.tsx

```diff
@@ -164,25 +184,33 @@ function SelectionTool(props: Props) {
      * - selection was not cancelled with Escape, and
      * - modifier key was released after pointer (if applicable). */
     if (prevSelection && hasSuccessfullyEndedRef.current) {
+      assertDefined(prevIsValid);
```

Comment:
`prevIsValid` is `undefined` on initial render, but since we know that `prevSelection` is defined, we know we're no longer in the initial render.

Reply:
Related to https://github.com/silx-kit/h5web/pull/1345/files#r1066859012: if you add `isValid to the Selection state, this `assert` may not be needed. This could give an argument in favour of this solution :shrug:

Reply:
I'm gonna keep it as is. Having a simple state makes all the event handling logic, as well the logical structure of the component, much simpler to understand. If a refactoring is needed in the future, I think it will be around those two `usePrevious` calls.